treeherder/model/models.py

```diff
@@ -1252,11 +1253,15 @@ class TextLogError(models.Model):
     """
 
     id = models.BigAutoField(primary_key=True)
-
-    step = models.ForeignKey(TextLogStep, on_delete=models.CASCADE, related_name='errors')
+    job = models.ForeignKey(Job, on_delete=models.CASCADE, related_name='text_log_error', null=True)
     line = models.TextField()
     line_number = models.PositiveIntegerField()
 
+    # TODO delete this field and unique_together once backfill of jobs in TextLogError table has been completed
+    step = models.ForeignKey(
+        TextLogStep, on_delete=models.CASCADE, related_name='errors', null=True
```

Comment:
Shouldn't we remove the `on_delete=models.CASCADE` here?  My understanding is that if we delete the `TextLogStep`s, then it will delete these `TextLogError`s if `CASCADE` is still set.

Reply:
The plan was to remove this entire foreign key relationship first (in a future pr), and then delete the TextLogStep table. Is there another order I should do this in?

Reply:
Oh, I see what you mean.  No, your order sounds good here.